src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,571 @@
+        @dev msg.sender must be authorized.
+        @param ilk  bytes32 of the D3M ilk to be updated
+        @param what bytes32("pool"|"plan") or it will revert
+        @param data address we are setting it to
+    */
+    function file(bytes32 ilk, bytes32 what, address data) external auth {
+        require(vat.live() == 1, "DssDirectDepositHub/no-file-during-shutdown");
+        require(ilks[ilk].tic == 0, "DssDirectDepositHub/pool-not-live");
+
+        if (what == "pool") ilks[ilk].pool = D3MPoolLike(data);
+        else if (what == "plan") ilks[ilk].plan = D3MPlanLike(data);
+        else revert("DssDirectDepositHub/file-unrecognized-param");
+        emit File(ilk, what, data);
+    }
+
+    // --- Deposit controls ---
+    function _wind(bytes32 ilk, D3MPoolLike pool, uint256 amount) internal {
+        // IMPORTANT: this function assumes Vat rate of this ilk will always be == 1 * RAY (no fees).
+        // That's why this module converts normalized debt (art) to Vat DAI generated with a simple RAY multiplication or division
+        // This module will have an unintended behaviour if rate is changed to some other value.
+        if (amount == 0) {
+            emit Wind(ilk, 0);
+            return;
+        }
+
+        require(int256(amount) >= 0, "DssDirectDepositHub/overflow");
+
+        vat.slip(ilk, address(pool), int256(amount));
+        vat.frob(ilk, address(pool), address(pool), address(this), int256(amount), int256(amount));
+        // normalized debt == erc20 DAI (Vat rate for this ilk fixed to 1 RAY)
+        daiJoin.exit(address(pool), amount);
+        pool.deposit(amount);
+
+        emit Wind(ilk, amount);
+    }
+
+    function _unwind(bytes32 ilk, D3MPoolLike pool, uint256 supplyReduction, uint256 availableAssets, Mode mode, uint256 assetBalance) internal {
+        // IMPORTANT: this function assumes Vat rate of this ilk will always be == 1 * RAY (no fees).
+        // That's why it converts normalized debt (art) to Vat DAI generated with a simple RAY multiplication or division
+        // This module will have an unintended behaviour if rate is changed to some other value.
+
+        EndLike end_;
+        uint256 daiDebt;
+        if (mode == Mode.NORMAL) {
+            // Normal mode or module just caged (no culled)
+            // debt is obtained from CDP art
+            (,daiDebt) = vat.urns(ilk, address(pool));
+        } else if (mode == Mode.MODULE_CULLED) {
+            // Module shutdown and culled
+            // debt is obtained from free collateral owned by this contract
+            // We rebalance the CDP after grabbing in `cull` so the gems represents
+            // the debt at time of cull
+            daiDebt = vat.gem(ilk, address(pool));
+        } else {
+            // MCD caged
+            // debt is obtained from free collateral owned by the End module
+            end_ = end;
+            end_.skim(ilk, address(pool));
+            daiDebt = vat.gem(ilk, address(end_));
```

Comment:
need to confirm the vat accounting in the scenario where this is called after someone pays back our debt.

After `skim` there should be the remainder of gem (art) in the d3m `urn` that needs to be cleared

Reply:
added test here: 609576b

@godsflaw @brianmcmichael @naszam My OCD wants to clear up that ink by moving it to  a gem balance and the `slip`ping it away.  But since at this point in shutdown it really doesn't matter because we will capture it in exec as fees when we unwind everything, I'm not sure its worth it. What do you think?